### PR TITLE
xblock-submit-and-compare | Replace pkg_resources to importlib_resources | Drop support for python 3.8

### PR DESCRIPTION
**Pull request by ttqureshi**
# Overview
This PR:
- replaces `pkg_resources` api from `importlib_resources` api
- drops support for `python 3.8` and add support for `python 3.11` & `python 3.12`
---
Tickets: 
- [Move on from deprecated pkg_resources api](https://github.com/openedx/xblock-submit-and-compare/issues/183)
- [Drop Python 3.8 & Add Support for Python 3.12](https://github.com/openedx/xblock-submit-and-compare/issues/186)

Followed [this](https://importlib-resources.readthedocs.io/en/latest/migration.html) migration guide.

## Testing
I have tested the changes against the following steps:
- Install and setup `xblock-submit-and-compare` in studio
- Tested the functionality of `xblock-submit-and-compare` in studio
- Run tox tests

## Screenshots
image

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ttqureshi!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads


#### :radio_button: Submit a signed contributor agreement (CLA)

:warning: We ask all contributors to the Open edX project to submit a [signed contributor agreement](https://openedx.org/cla) or indicate their institutional affiliation.
Please see the [CONTRIBUTING](https://github.com/openedx/.github/blob/master/CONTRIBUTING.md) file for more information.

If you've signed an agreement in the past, you may need to re-sign.
See [The New Home of the Open edX Codebase](https://open.edx.org/blog/the-new-home-of-the-open-edx-codebase/) for details.

Once you've signed the CLA, please allow 1 business day for it to be processed.
After this time, you can re-run the CLA check by adding a comment below that you have signed it.
If the CLA check continues to fail, you can tag the `@openedx/cla-problems` team in a comment for further assistance.






#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?



This repository is currently unmaintained.

To get help with finding a technical reviewer, tag the community contributions project manager for this PR in a comment and let them know that your changes are ready for review:

1. On the right-hand side of the PR, find the Contributions project, click the caret in the top right corner to expand it, and check the "Primary PM" field for the name of your PM.
2. Find their GitHub handle [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3548807177/Community+Contributions+Project+Manager#Current-OSPR-Project-Managers).


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

**Comment by farhan** [comment]
Please update major version as we have dropped support of Python 3.8

https://github.com/openedx/xblock-submit-and-compare/pull/187/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R10

File: submit_and_compare/mixins/scenario.py [hunk 1]
```diff
@@ -66,7 +67,7 @@ def workbench_scenarios(cls):
         """
         module = cls.__module__
         module = module.split('.', maxsplit=1)[0]
-        directory = pkg_resources.resource_filename(module, 'scenarios')
+        directory = importlib.resources.files(module) / 'scenarios'
```
Comment (farhan) [inline submit_and_compare/mixins/scenario.py:70]: We are performing different file operations in this python module. We should update all of them as per following migration guide to keep the code optimized.

https://importlib-resources.readthedocs.io/en/latest/migration.html

File: CHANGELOG.md [hunk 2]
```diff
@@ -46,3 +46,7 @@ Version 2.0.0
 Version 2.1.0
 
                 Added support for python 3.11 support. Removed django 3.2 support.
+
+Version 2.2.0
```
Comment (farhan) [inline CHANGELOG.md:50]: We should update major version as we are dropping support of python version

Comment (ttqureshi) [reply CHANGELOG.md:50]: ok got it

File: submit_and_compare/mixins/scenario.py [hunk 3]
```diff
@@ -64,9 +21,4 @@ def workbench_scenarios(cls):
         """
         Gather scenarios to be displayed in the workbench
         """
-        module = cls.__module__
-        module = module.split('.', maxsplit=1)[0]
-        directory = pkg_resources.resource_filename(module, 'scenarios')
-        files = _find_files(directory)
-        scenarios = _read_files(files)
-        return scenarios
+        return loader.load_scenarios_from_path("../scenarios")
```
Comment (farhan) [inline submit_and_compare/mixins/scenario.py:24]: Add extra line at the end of file

**Comment by farhan** [comment]
> Please update major version as we have dropped support of Python 3.8
> 
> https://github.com/openedx/xblock-submit-and-compare/pull/187/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R10

Please address this

**Review by farhan — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -46,3 +46,7 @@ Version 2.0.0
 Version 2.1.0
 
                 Added support for python 3.11 support. Removed django 3.2 support.
+
+Version 3.0.0
+
+		Dropped support for Python 3.8 and added support for Python 3.12.
```
Comment (farhan) [inline CHANGELOG.md:52]: Dropped support for Python 3.8 and added support for Python 3.11 and 3.12.

**Review by farhan — APPROVED** [review]

**Review by feanil — APPROVED** [review]

**Comment by ttqureshi** [comment]
resolved #183 
resolved #186

**Pull request merged**